### PR TITLE
Add flag for subscription based emails for custom user target

### PR DESCRIPTION
**Pull request by rish9101**

### CODE REVIEW

File: actions.py [hunk 1]
```diff
@@ -56,11 +60,19 @@ def email_push(
 
     if has_custom_user_target:
         if persons is not None:
+
+            if send_only_to_subscribed_users:
+                persons = UserSubscription.objects.filter(
```
Comment (pradumangoyal) [inline actions.py:65]: Don't we need to filter based on action also? See line number 86.

Comment (rish9101) [reply actions.py:65]: Done now.

Comment (pradumangoyal) [reply actions.py:65]: same for emails.

File: actions.py [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
-from django.core.mail import EmailMessage
 from django.conf import settings
-from email.mime.image import MIMEImage
 
-from kernel.models import Person
 from categories.redisdb import Subscription
+
 from emails.html_content import get_html_content
 from emails.utils.get_people_contact import get_people_contact
 from emails.tasks.push_email import queue_push
```
Comment (pradumangoyal) [inline actions.py:12]: This line is extra I think

**Pull request merged**